### DISCUSSION TITLE
Dist & BodyDist Trigger

### DISCUSSION THREAD

**Discussion by GodDestroya** [ikemen-engine/Ikemen-GO#1980] [Feature Request]
### Is your feature request related to a problem? Please describe.

I wanted to implement a Tranform into Kyubi Susano onto my Naruto char only on Team Mode and with Sasuke as partner. To make it more realist It would have been very interesting to verify Sasuke is near Naruto during the transformation, but the only distance triggers redirect to the opponent (P2). 

### Describe the solution you'd like

I think either P2Dist and P2BodyDist should just be Dist and BodyDist and accompanied by a trigger redirection to let know who should be targeted (the Best solution according to me)

Or you should just add Dist and P2BodyDist triggers for all the player

### Search terms

distance trigger

**Comment by potsmugen** [comment]
You can do this with some simple math. Dist:

`(playerID(ID), pos x - pos x) * facing`

BodyDist:

`(playerID(ID), pos x - pos x - playerID(ID), const(size.ground.front) - const(size.ground.front)) * facing`

Or something like that. Untested. The latter may need sign adjustments. And front/back variation if you want to be really precise.

However your suggestion isn't without merit.
